src/lazy/decoder.rs

```diff
@@ -60,13 +74,73 @@ pub trait Decoder: 'static + Sized + Debug + Clone + Copy {
 }
 
 pub trait RawVersionMarker<'top>: Debug + Copy + Clone + HasSpan<'top> {
```

Comment:
I don't think this is as clear-cut as that.

> Personally, I would get rid of the new_ prefix so that only "old" functions are prefixed, but I don't have a strong preference how you do it so long as it's consistent.

If someone calls `ivm.encoding()`, I think they could reasonably expect that to mean the encoding that was used to encode the IVM. However, under this proposal it actually means the encoding that the _next_ stream segment will use.

The methods which do not have a prefix are:
* `is_text`
* `is_binary`
* `major_minor`

`is_text` and `is_binary` have no concept of "new"/"old" -- they MUST agree for the stream segment leading up to the IVM and the stream segment that follows it.

`major_minor` is the major-minor version being marked by the IVM. I'm also having a hard time finding a way to misinterpret this one--if there were new/old versions of it, they would also be forced to always agree.

I could get behind changing the affected names to refer to the stream segments, i.e.:
* `stream_encoding_before`/`stream_encoding_after`
* `stream_version_before`/`stream_version_after`
* `is_text`/`is_binary`
* `major_minor`

What do you think of that?

Reply:
> major_minor is the major-minor version being marked by the IVM. I'm also having a hard time finding a way to misinterpret this one--if there were new/old versions of it, they would also be forced to always agree.

I think I was interpreting this as being the `major_minor` of the Ion encoding (which one—before or after?) rather than of the IVM and that's why I was getting confused. 

I think that adding `_before` and `_after` is a very helpful clarification (and it's easier to catch the implied "right now" that exists between "before" and "after").